library/src/main/java/com/instacart/truetime/time/TrueTimeImpl.kt

```diff
@@ -0,0 +1,173 @@
+package com.instacart.truetime.time
+
+import com.instacart.truetime.TrueTimeEventListener
+import com.instacart.truetime.NoOpEventListener
+import com.instacart.truetime.sntp.Sntp
+import com.instacart.truetime.sntp.SntpImpl
+import com.instacart.truetime.time.TrueTimeParameters.Builder
+import java.net.Inet6Address
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import java.net.InetAddress
+import java.net.UnknownHostException
+import java.util.Date
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.async
+import kotlinx.coroutines.cancelChildren
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.selects.select
+
+class TrueTimeImpl(
```

Comment:
Since this is the one people will use, I think it might be clearer to name this one `TrueTime` and name the interface `TrueTimeBase` or something similar, so that `TrueTime` is the more discoverable one

Reply:
hmm... i prefer having `TrueTime` be the interface mostly cause that's what people will be inclined to use and pass around more. It being an interface means they will use the interface in most places which is what we want 🙂.  `TrueTimeImpl` should probably just be used in one place and forgotten.